### PR TITLE
feat: domain lookup in sso loging screen

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

[WPB-255](https://wearezeta.atlassian.net/browse/WPB-255)

### Solutions

do domain lookup in sso login screen if the user input is an emil

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

[WPB-255]: https://wearezeta.atlassian.net/browse/WPB-255?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1847/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1847/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6727474)(6727474)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6727474)(6727474)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1847/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6778002)(6778002)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6778002)(6778002)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-1847/7/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6795311)(6795311)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6795311)(6795311)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/authentication/login/sso/LoginSSOViewModel.kt [hunk 1]
```diff
@@ -60,13 +68,109 @@ class LoginSSOViewModel @Inject constructor(
...
+                            it.authenticationScope
+                        }
+                    }
+                }
+
+                authScope.ssoLoginScope.fetchSSOSettings().also {
+                    when (it) {
+                        is FetchSSOSettingsUseCase.Result.Failure -> {}
+                        is FetchSSOSettingsUseCase.Result.Success -> {
+                            it.defaultSSOCode?.let { ssoCode ->
```
Comment (ohassine) [inline app/src/main/kotlin/com/wire/android/ui/authentication/login/sso/LoginSSOViewModel.kt:117]: Can we cover error cases with test as well ? when `AutoVersionAuthScopeUseCase` `fetchSSOSettings` fail

File: app/src/main/kotlin/com/wire/android/ui/authentication/login/sso/LoginSSOViewModel.kt [hunk 2]
```diff
@@ -60,13 +68,109 @@ class LoginSSOViewModel @Inject constructor(
...
+                        }
+
+                        is AutoVersionAuthScopeUseCase.Result.Success -> it.authenticationScope
+                    }
+                }
+
+            defaultAuthScope.domainLookup(loginState.userInput.text).also {
+                when (it) {
+                    is DomainLookupUseCase.Result.Failure -> {
+                        loginState = loginState.copy(ssoLoginLoading = false, loginError = it.toLoginError())
```
Comment (ohassine) [inline app/src/main/kotlin/com/wire/android/ui/authentication/login/sso/LoginSSOViewModel.kt:158]: Same thing, can we cover failure case with test ?

**Comment by AndroidBob** [comment]
Build [10](https://10.10.124.134/job/wire-android-reloaded/job/PR-1847/10/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6828412)(6828412)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6828412)(6828412)-staging-compat.apk)

**Review by saleniuk — APPROVED** [review]

**Review by ohassine — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [183](https://10.10.124.134/job/AR-build-pipeline/183/) **failed**.

**Comment by AndroidBob** [comment]
Build [187](https://10.10.124.134/job/AR-build-pipeline/187/) **failed**.

**Comment by AndroidBob** [comment]
Build [196](https://10.10.124.134/job/AR-build-pipeline/196/) **failed**.

**Comment by AndroidBob** [comment]
Build [210](https://10.10.124.134/job/AR-build-pipeline/210/) **failed**.

**Comment by AndroidBob** [comment]
Build [211](https://10.10.124.134/job/AR-build-pipeline/211/) **failed**.

**Comment by AndroidBob** [comment]
Build [221](https://10.10.124.134/job/AR-build-pipeline/221/) **failed**.

**Comment by AndroidBob** [comment]
Build [252](https://10.10.124.134/job/AR-build-pipeline/252/) **failed**.

**Comment by AndroidBob** [comment]
Build [256](https://10.10.124.134/job/AR-build-pipeline/256/) **failed**.

**Comment by AndroidBob** [comment]
Build [264](https://10.10.124.134/job/AR-build-pipeline/264/) **failed**.

**Comment by AndroidBob** [comment]
Build [270](https://10.10.124.134/job/AR-build-pipeline/270/) **failed**.

**Comment by AndroidBob** [comment]
Build [315](https://10.10.124.134/job/AR-build-pipeline/315/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.3.0(7010512)(7010512)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.3.0(7010512)(7010512)-staging-compat.apk)

**Pull request merged**